moveit_ros/move_group/src/default_capabilities/move_action_capability.cpp

```diff
@@ -64,20 +65,26 @@ void move_group::MoveGroupMoveAction::executeMoveCallback(const moveit_msgs::Mov
   context_->planning_scene_monitor_->waitForCurrentRobotState(ros::Time::now());
   context_->planning_scene_monitor_->updateFrameTransforms();
 
+  // Update the goal's start state to the robot's current state using the latest data from the planning scene
+  planning_scene_monitor::LockedPlanningSceneRO lscene(context_->planning_scene_monitor_);
   moveit_msgs::MoveGroupResult action_res;
-  if (goal->planning_options.plan_only || !context_->allow_trajectory_execution_)
+  const robot_state::RobotState& current_state = lscene->getCurrentState();
+  moveit_msgs::MoveGroupGoal* new_goal = new moveit_msgs::MoveGroupGoal(*goal);
+  robot_state::robotStateToRobotStateMsg(current_state, new_goal->request.start_state);
+  moveit_msgs::MoveGroupGoalConstPtr change_goal(new_goal);
```

Comment:
Why do you create change_goal here? Couldn't you just create a non-const shared pointer to the new goal once up above as:
```
moveit_msgs::MoveGroupGoalPtr new_goal(new moveit_msgs::MoveGroupGoal(*goal))
```
And then use that from then on (never create a `change_goal` variable?

Reply:
I will be revising the code and tests in the weekend. Thanks for feedback.

Reply:
Suggested changes are made.